app/controllers/training_tips_controller.rb

```diff
@@ -0,0 +1,18 @@
+class TrainingTipsController < ApplicationController
+  before_action :authorize!
```

Comment:
this needs to also be checking user authorization, so you should extend the `WithUserController` instead to provide that functionality for you

Reply:
done